subsys/metrics/src/main/java/org/commonjava/indy/metrics/zabbix/sender/ZabbixSender.java

```diff
@@ -0,0 +1,176 @@
+package org.commonjava.indy.metrics.zabbix.sender;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Created by xiabai on 4/7/17.
+ * Copy from
+ */
+public class ZabbixSender
+{
+    private static final Pattern PATTERN = Pattern.compile( "[^0-9\\.]+");
+    private static final Logger logger = LoggerFactory.getLogger( ZabbixSender.class );
+    String host;
+    int port;
+    int connectTimeout = 3 * 1000;
+    int socketTimeout = 3 * 1000;
+
+    public ZabbixSender(String host, int port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    public ZabbixSender(String host, int port, int connectTimeout, int socketTimeout) {
+        this(host, port);
+        this.connectTimeout = connectTimeout;
+        this.socketTimeout = socketTimeout;
+    }
+
+    public SenderResult send(DataObject dataObject) throws IOException
+    {
+        return send(dataObject, System.currentTimeMillis() / 1000);
+    }
+
+    /**
+     *
+     * @param dataObject
+     * @param clock
+     *            TimeUnit is SECONDS.
+     * @return
+     * @throws IOException
+     */
+    public SenderResult send(DataObject dataObject, long clock) throws IOException {
+        return send( Collections.singletonList( dataObject), clock);
+    }
+
+    public SenderResult send(List<DataObject> dataObjectList) throws IOException {
+        return send(dataObjectList, System.currentTimeMillis() / 1000);
+    }
+
+    /**
+     *
+     * @param dataObjectList
+     * @param clock
+     *            TimeUnit is SECONDS.
+     * @return
+     * @throws IOException
+     */
+    public SenderResult send(List<DataObject> dataObjectList, long clock) throws IOException {
```

Comment:
Depending on how many concurrent requests we need to make to the Zabbix server, we might need to port this Socket implementation over to XNIO, which is a much higher-performing library doing non-blocking socket I/O. I don't think we need to do it for now, but we should put in a //TODO for this method to explore XNIO.